trends_presentation/src/commonMain/kotlin/net/thechance/mena/trends/presentation/screen/show_real/TrendsScreen.kt

```diff
@@ -0,0 +1,135 @@
+import org.jetbrains.compose.resources.stringResource
+import org.koin.compose.viewmodel.koinViewModel
+
+@Composable
+internal fun TrendsScreen(
+    viewModel: TrendsViewModel = koinViewModel()
+) {
+    val state by viewModel.state.collectAsStateWithLifecycle()
+    val navController = LocalNavController.current
+
+    ObserveAsEffect(viewModel.effect) { effect ->
+        when (effect) {
+            is TrendsUiEffect.NavigateToReelDetails ->
+                navController.navigate(Route.ReelDetails(effect.trendId))
+            is TrendsUiEffect.NavigateToAddReel ->
+                navController.navigate(Route.UploadReel)
+            is TrendsUiEffect.NavigateToChangeTags ->
+                navController.navigate(Route.Categories)
+            is TrendsUiEffect.NavigateToManageTrends ->
+                navController.navigate(Route.ManageReels)
+        }
+    }
+
+    TrendsScreenContent(
+        state = state,
+        listener = viewModel,
+    )
+}
+
+@Composable
+private fun TrendsScreenContent(
+    state: TrendsScreenState,
+    listener: TrendsInteractionListener,
+) {
+    val reels = state.reels.collectAsLazyPagingItems()
+
+    Scaffold(
+        topBar = {
+            AppBar(
+                title = stringResource(Res.string.trends_title),
+                trailingContent = {
+                    AppBarOptionContainer(isBadgeVisible = false, onClick = listener::onManageTrendsClick) {
+                        Icon(
+                            painter = painterResource(Res.drawable.ic_account_setting),
+                            contentDescription = stringResource(Res.string.manage_trends),
+                            tint = Theme.colorScheme.shadePrimary
+                        )
+                    }
+                    AppBarOptionContainer(isBadgeVisible = false, onClick = listener::onEditTagsClick) {
+                        Icon(
+                            painter = painterResource(Res.drawable.ic_pencil_edit),
+                            contentDescription = stringResource(Res.string.edit_tags),
+                            tint = Theme.colorScheme.shadePrimary
+                        )
+                    }
+                }
+            )
+        }
+    ) {
+        Box(modifier = Modifier.fillMaxSize()) {
+            LazyColumn(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(horizontal = Theme.spacing._16, vertical = Theme.spacing._8),
+                verticalArrangement = Arrangement.spacedBy(Theme.spacing._16)
+            ) {
+                items(reels.itemSnapshotList.items) { reel ->
+                    reel.let { reel ->
+                        TrendCard(
+                            trend = reel,
+                            onMoreClick = { listener::onMoreClick },
+                            onLikeClick = { listener.onLikeClick(reel.id) },
+                            onVideoClick = { listener.onVideoClick(reel.id) }
```

Comment:
Can we make this to use reflection in all the screen, or this way more much complex and it is not necessary to use it?

```suggestion
                onLikeClick = { listener::onLikeClick.invoke(reel.id) },
                 onVideoClick = { listener::onVideoClick.invoke(reel.id) }